test/distributed/_tools/test_runtime_estimator.py

```diff
@@ -0,0 +1,199 @@
+    @unittest.skipIf(not TEST_CUDA, "CUDA not available")
+    def test_transformer_runtime(
+        self,
+    ):
+        """Runs a basic GPT-2 model"""
+        vocab_size = 8192
+        bsz, seq_len = 64, 1024
+        model_args = ModelArgs(
+            n_layers=4,
+            n_heads=12,
+            vocab_size=vocab_size,
+            max_seq_len=seq_len,
+            dim=768,
+            dropout_p=0.1,
+        )
+
+        args = self._init_model_and_args("Transformer", model_args, bsz)
+        actual_runtime = self._measure_actual_cuda_time(self._train_step, args)
+
+        with FakeTensorMode():
+            fake_args = self._init_model_and_args("Transformer", model_args, bsz)
+            benchmark_estimate = self._runtime_estimate(
+                "operator-level-benchmark", self._train_step, fake_args
+            )
+            roofline_estimate = self._runtime_estimate(
+                "operator-level-cost-model", self._train_step, fake_args
+            )
+        benchmark_accuracy = actual_runtime / benchmark_estimate
+        roofline_accuracy = actual_runtime / roofline_estimate
+        print(
+            f"Actual: {actual_runtime} Benchmark Estimate: {benchmark_estimate} Accuracy: {benchmark_accuracy}"
+            f"\n Actual: {actual_runtime} Roofline Estimatee: {roofline_estimate} Accuracy: {roofline_accuracy}"
+        )
+        self.assertAlmostEqual(benchmark_accuracy, 1.0, delta=0.2)
+        self.assertAlmostEqual(roofline_accuracy, 1.0, delta=0.3)
+
+    @skipIfTorchDynamo("https://github.com/pytorch/pytorch/issues/115653")
+    @unittest.skipIf(not TEST_CUDA, "CUDA not available")
+    def test_conv_model_runtime(
+        self,
+    ):
+        """Runs a simple CNN model"""
+        num_classes = 1024
+        bsz, img_sz = 256, 512
+        model_args = ConvArgs(img_sz, num_classes)
+        args = self._init_model_and_args("CNN", model_args, bsz)
+        actual_runtime = self._measure_actual_cuda_time(self._train_step, args)
+
+        with FakeTensorMode():
+            fake_args = self._init_model_and_args("CNN", model_args, bsz)
+            benchmark_estimate = self._runtime_estimate(
+                "operator-level-benchmark", self._train_step, fake_args
+            )
+            roofline_estimate = self._runtime_estimate(
+                "operator-level-cost-model", self._train_step, fake_args
+            )
+        benchmark_accuracy = actual_runtime / benchmark_estimate
+        roofline_accuracy = actual_runtime / roofline_estimate
+        print(
+            f"Actual: {actual_runtime} Benchmark Estimate: {benchmark_estimate} Accuracy: {benchmark_accuracy}\n"
+            f"Actual: {actual_runtime} Roofline Estimatee: {roofline_estimate} Accuracy: {roofline_accuracy}"
+        )
+        self.assertAlmostEqual(benchmark_accuracy, 1.0, delta=0.2)
+        self.assertAlmostEqual(roofline_accuracy, 1.0, delta=0.4)
```

Comment:
The conv-estimates are not that great with the roofline model. The main goal of the test is to show that the code runs for all kinds of models. We need a learned estimate for conv kernels.